### ISSUE TITLE
bug(Material): Checkbox has border bottom with different opacity 

### ISSUE THREAD

**Issue by jimmykane** [angular/components#20402]
#### Reproduction

Steps to reproduce:

1. Change the background of a checkbox, but basically look at the source code linked below. This CSS rule should not belong there as it has side effects 
 
 
#### Expected Behavior

Consident border color for checkboxes

#### Actual Behavior

Bottom border is less vivid (opacity)
<img width="74" alt="Screenshot 2020-08-24 at 21 21 40" src="https://user-images.githubusercontent.com/1268723/91087042-1c9c5100-e650-11ea-8e90-ae82ca21bd8c.png">

#### Environment
```
"@angular/cdk": "^10.1.3",
    "@angular/common": "10.0.9",
    "@angular/compiler": "10.0.9",
    "@angular/core": "10.0.9",
    "@angular/fire": "6.0.0",
    "@angular/forms": "10.0.9",
    "@angular/material": "^10.1.3",
```

Issue is due to a solo "weird" css rule in https://github.com/angular/components/blob/227e568d3177886407d8c168d67a148cc6383c01/src/material/checkbox/_checkbox-theme.scss#L26 

```
  .mat-checkbox-frame {
    border-color: mat-color($foreground, secondary-text);
  }
```

Why only border bottom this "change"?

**Comment by crisbeto** [comment]
Could it be an override from somewhere else in your app? I'm not seeing it against the code in master:

![Angular_Material_-_Google_Chrome_2020-08-24_21-41-44](https://user-images.githubusercontent.com/4450522/91088676-bbc24800-e652-11ea-9593-477400f11534.png)

**Comment by jimmykane** [comment]
I will double check and let you know

**Comment by jimmykane** [comment]
Hi @devversion 

I am adding 

```
background-color: rgb(205, 248, 0);
```

to 

```
<mat-checkbox>
````


The checkbox is inside a material table

**Comment by jimmykane** [comment]
Here is the first Error: 

Background overlaps the border: 

https://angular-gmbl1c.stackblitz.io

<img width="33" alt="Screenshot 2020-08-25 at 12 12 56" src="https://user-images.githubusercontent.com/1268723/91162410-5743cf00-e6cc-11ea-9d1e-8819f9a1a07d.png">


Then I adjust the container to become bigger: 
```
::ng-deep  .mat-checkbox-inner-container-no-side-margin{
  width: 17px !important;
  height: 17px !important;
}
```

And this is then exposed: 

https://angular-gmbl1c-y3hncs.stackblitz.io 

<img width="33" alt="Screenshot 2020-08-25 at 12 15 19" src="https://user-images.githubusercontent.com/1268723/91162653-aa1d8680-e6cc-11ea-9ba2-d6b3df962f35.png">


So basically I cannot change the background of the checkbox inside a material table without having visual issues. 

The reason is the part of code I linked on the first comment / issue  

IS that enough @devversion ? 

Thank you very much for looking into this

**Comment by crisbeto** [comment]
The problem isn't that the bottom border is different, but that it sticks out underneath the background which makes it look different since the `rgba` color causes the different to mix. I don't think we can do much about it since you're overriding our built-in styles and accommodating for all the different overrides isn't practical. For your case, maybe it can be worked around by wrapping the checkbox in a `div` and styling its background instead?

**Comment by jimmykane** [comment]
This is not an override of a build in style. Except if you think adding a color is an override. Then we should not color any component right? 

I am just adding background on an Angular component that works but not on table rows. I am not doing anything else. The rest of the code I posted as an example was my try to fix this on my side. 

@crisbeto  you have a specific RGA opacity change for the bottom border: 
```
.mat-checkbox-frame {
    border-color: mat-color($foreground, secondary-text);
  }
```

https://github.com/angular/components/blob/227e568d3177886407d8c168d67a148cc6383c01/src/material/checkbox/_checkbox-theme.scss#L26 

That causes this issue. 

Why is that css rule there?  

For reference here is just adding the background color: https://angular-gmbl1c.stackblitz.io nothing more

**Comment by crisbeto** [comment]
The CSS rule is there, because the checkbox frame has to have a border color. As I mentioned above, the color is the same on all sides, but the bottom one looks different because it's on top of a white background, instead of the red one. If you want the entire frame to be inside the red background, you can add this to your styles:

```css
.mat-checkbox-layout {
  vertical-align: top;
}
```

**Comment by jimmykane** [comment]
Alright ok. This explains that. Thanks. 

should we close this one ?

**Comment by crisbeto** [comment]
No problem and yeah I'll close the issue.

**Comment by jimmykane** [comment]
One question, sorry but I tried the solution with no effect. Then the top part of the border is outside....
